views/pages/fixture_editor.js

```diff
@@ -784,147 +888,181 @@ function getSubmitDialogString() {
+ * @param {!string} modelKey The Vue reference used in v-model.
+ * @param {!InputOptions} options Additional optional parameters.
  * @returns {!string} The input element HTML, possibly together with related elements.
  */
-function textInput(key, property, hint, id, additionalAttributes = ``) {
+function textInput(modelKey, options = {}) {
   let html = `<input type="text"`;
-  html += getRequiredAttr(property);
-
-  if (property.enum) {
-    html += ` list="${id}-list"`;
-  }
-
-  html += getPlaceholderAttr(hint);
-  html += additionalAttributes;
-  html += ` v-model="${key}" />`;
-
-  if (property.enum) {
-    html += `<datalist id="${id}-list">`;
-    for (const item of property.enum) {
-      html += `<option>${item}</option>`;
-    }
-    html += `</datalist>`;
-  }
+  html += getRequiredAttr(options.required);
+  html += getPlaceholderAttr(options.hint);
+  html += getPatternAttr(options.property);
 
+  html += getAdditionalAttributes(options.attributes);
+  html += ` v-model="${modelKey}" />`;
   return html;
 }
 
 /**
- * @param {!string} key The key in Vue's data model that this URL input element is bound to.
- * @param {*} property The JS Schema property that this input element has to obey.
- * @param {?string} hint A hint to the user that helps filling the right value into this input element.
+ * @param {!string} modelKey The Vue reference used in v-model.
+ * @param {!InputOptions} options Additional optional parameters.
  * @returns {!string} The input element HTML.
  */
-function urlInput(key, property, hint) {
+function urlInput(modelKey, options = {}) {
   let html = `<input type="url"`;
-  html += getRequiredAttr(property);
-  html += getPlaceholderAttr(hint);
-  html += ` v-model="${key}" />`;
+  html += getRequiredAttr(options.required);
+  html += getPlaceholderAttr(options.hint);
+  html += getAdditionalAttributes(options.attributes);
+  html += ` v-model="${modelKey}" />`;
   return html;
 }
 
 /**
- * @param {!string} key The key in Vue's data model that this textarea element is bound to.
- * @param {*} property The JS Schema property that this textarea element has to obey.
- * @param {?string} hint A hint to the user that helps filling the right value into this textarea element.
+ * @param {!string} modelKey The Vue reference used in v-model.
+ * @param {!InputOptions} options Additional optional parameters.
  * @returns {!string} The textarea element HTML.
  */
-function textareaInput(key, property, hint) {
+function textareaInput(modelKey, options = {}) {
   let html = `<textarea`;
-  html += getRequiredAttr(property);
-  html += getPlaceholderAttr(hint);
-  html += ` v-model="${key}"></textarea>`;
+  html += getRequiredAttr(options.required);
+  html += getPlaceholderAttr(options.hint);
+  html += getPatternAttr(options.property);
+  html += (options.property && options.property.minLength) ? ` minlength="${options.property.minLength}"` : ``;
```

Comment:
This makes it impossible to leave the textarea empty if it is not required. Better just remove this line.

Reply:
No – the field is only invalid, if anything has been entered and that length is below `minlength`.

Currently, this is only used with comments as a minlength of 1, which has no effect at all. However, a minlength of 5 would forbid "abc", but not an empty field.

It's [not natively supported](https://caniuse.com/#search=minlength) in neither IE nor Edge. Does that matter?

Reply:
Ah, then it's good to have! We use a validate polyfill anyway, so missing IE and Edge support is no problem.